### PR TITLE
Updated PDAs 

### LINKED ISSUE #273: Atmos PDA: provide atmos alert monitoring
Just like power monitoring in engineering PDA cartridge, so atmos wouldn't have to run back and forth just to see alerts.

### PR DESCRIPTION
**Pull request by Cruix**
Fixes #273 
*fixed some cases where the botanist's PDA would not act like a plant analyzer when it should. Removed some copy-pasted code.
*PDA cartridges can now receive atmos, fire, and power alerts. Atmos techs receive atmos and fire, Engineers receive power, and the CE and Captain receive all 3. Alert notifications can be toggled on, causing the PDA to beep and state alerts when they are received.
*Atmos tech, CE, and Captain cartridges can view all atmos monitoring consoles through their PDA, similar to how engineers can view power monitoring consoles.
*Changed scanmode magic numbers into #defines.
*Engineer PDA cartridges have a new scanmode that lets them measure power in wires like a multitool.
*Added some new menu categories, moved some options out of "Utilities".
*Captain's Value-PAK cartridge now has 100% more value.

I have tested this quite thoroughly, but I messed with so much bad PDA code that I would prefer lots of additional testing before it is pushed to live.

### CODE REVIEW

File: code/game/area/areas.dm [hunk 1]
```diff
@@ -72,10 +77,17 @@
 					D.cancelAlarm("Power", src, source)
 				else
 					D.triggerAlarm("Power", src, cameras, source)
+			for(var/listener in power_alert_listeners)
+				if(state == 1)
+					listener:cancelAlarm("Power", src, source)
+				else
+					listener:triggerAlarm("Power", src, cameras, source)
```
Comment (X-TheDark) [inline code/game/area/areas.dm]: Please don't use this style. WHen I meant typeless loops, all I meant is
for(var/x in something)
var/typecast/here/please/variable = x

**Comment by Alek2ander** [comment]
AI antags are unhappy

**Comment by DavidEdwards** [comment]
What do you mean Alek?

**Comment by DavidEdwards** [comment]
Please also make sure that you add a changelog .yml file indicating your changes. There is an example here: /html/changelogs/example.yml. (unless your recent PR contained that already)

**Comment by Alek2ander** [comment]
> What do you mean Alek?

Right now antag AIs that do funny stuff (syphon rooms, vent plasma, etc) can get away with it because only supernerds check alerts on the console. But if every engineer has a PDA screaming at them, AI has to change the air alarm alert settings (which is clunky to do) if it wants to maintain any sort of stealth.

**Comment by DavidEdwards** [comment]
I see. That is a good point. Worth consideration.

File: code/game/area/areas.dm [hunk 2]
```diff
@@ -72,10 +77,17 @@
 					D.cancelAlarm("Power", src, source)
 				else
 					D.triggerAlarm("Power", src, cameras, source)
+			for(var/listener in power_alert_listeners)
+				if(state == 1)
+					listener:cancelAlarm("Power", src, source)
```
Comment (X-TheDark) [inline code/game/area/areas.dm:82]: Untyped variables.
SHUN! SHUUUUUUUUUUUNNNNNNNNNNNNNNNNNN!

Comment (Cruix) [inline code/game/area/areas.dm:82]: HUUUUUUUUSH this is one area where it makes sense to use the : operator, right now the alert_listener lists only contain cartridges, but they could very well contain everything in all 3 of those other horribly unmodular for-loops.

**Comment by Cruix** [comment]
> Right now antag AIs that do funny stuff (syphon rooms, vent plasma, etc) can get away with it because only supernerds check alerts on the console

I don't think we should leave the game in such a state that antags think they can get away with things because most players are too lazy to do their job. What if traitors were willing to combo-cafe their target in the central hallway with plenty of witnesses because they know that sec is too incompetent to listen to their radio? But, if everyone disagrees with me, it will be easy enough to just remove the flags from everyone's PDAs.

I considered having alerts go through the PDA server, so the AI could turn that off, but decided against that because turning the PDA server off requires that the AI's camera go all the way to tcomms, which itself is very clunky.

**Comment by Cruix** [comment]
I accidentally closed it...

**Comment by DavidEdwards** [comment]
I think that we should go ahead with this, and see how it goes. We can always disable parts of it if it is too much.

**Pull request merged**